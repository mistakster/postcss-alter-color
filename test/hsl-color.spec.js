const postcss = require('postcss');
const alterColorPlugin = require('../lib/index');

it('should alter hsl colors', () => {
  return postcss()
    .use(alterColorPlugin({from: 'black', to: 'red'}))
    .process(`div{color:hsl(0,0%,0%)}`, {from: undefined})
    .then(result => {
      expect(result.css).toBe(`div{color:hsl(0,100%,50%)}`)
    });
});

it('should alter hsl color in a complex value', () => {
  return postcss()
    .use(alterColorPlugin({from: 'black', to: 'red'}))
    .process(`div{color:hsl(0,0%,0%);border:1px solid hsl(0,0%,100%);outline:1px solid hsl(0,0%,0%)}`, {from: undefined})
    .then(result => {
      expect(result.css).toBe(`div{color:hsl(0,100%,50%);border:1px solid hsl(0,0%,100%);outline:1px solid hsl(0,100%,50%)}`)
    });
});
