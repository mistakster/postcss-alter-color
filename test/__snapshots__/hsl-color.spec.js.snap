// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should alter hsl color in a complex value 1`] = `
"
div {
  color: hsl(0,100%,50%);
  border: 1px solid hsl(0,0%,100%);
  outline: 1px solid hsl(0,100%,50%);
}
"
`;

exports[`should alter hsl colors 1`] = `
"
div {
  color: hsl(0,100%,50%);
}
"
`;
